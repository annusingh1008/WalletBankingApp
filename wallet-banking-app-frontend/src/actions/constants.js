export const authConstants = {
    LOGIN_REQUEST: 'LOGIN_REQUEST',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGIN_FAILURE: 'LOGIN_FAILURE',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    LOGOUT_FAILURE: 'LOGOUT_FAILURE'
}

export const userConstants = {
    USER_REGISTER_REQUEST: 'USER_REGISTER_REQUEST',
    USER_REGISTER_SUCCESS: 'USER_REGISTER_SUCCESS',
    USER_REGISTER_FAILURE: 'USER_REGISTER_FAILURE',
}

export const amountTransferConstants = {
    AMOUNT_TRANSFER_REQUEST: 'AMOUNT_TRANSFER_REQUEST',
    AMOUNT_TRANSFER_SUCCESS: 'AMOUNT_TRANSFER_SUCCESS',
    AMOUNT_TRANSFER_FAILURE: 'AMOUNT_TRANSFER_FAILURE'
}

export const rechargeConstants = {
    RECHARGE_REQUEST: 'RECHARGE_REQUEST',
    RECHARGE_SUCCESS: 'RECHARGE_SUCCESS',
    RECHARGE_FAILURE: 'RECHARGE_FAILURE',
}

export const getUserDetailsConstants = {
    GET_USER_DETAILS_REQUEST: 'GET_USER_DETAILS_REQUEST',
    GET_USER_DETAILS_SUCCESS: 'GET_USER_DETAILS_SUCCESS',
    GET_USER_DETAILS_FAILURE: 'GET_USER_DETAILS_FAILURE'
}

export const transactionsConstants = {
    TRANSACTIONS_REQUEST: 'TRANSACTIONS_REQUEST',
    TRANSACTIONS_SUCCESS: 'TRANSACTIONS_SUCCESS',
    TRANSACTIONS_FAILURE: 'TRANSACTIONS_FAILURE',
    TOTAL_TRANSACTIONS_SUCCESS: 'TOTAL_TRANSACTIONS_SUCCESS',
}

export const cashbackConstants = {
    CASHBACK_REQUEST: 'CASHBACK_REQUEST',
    CASHBACK_SUCCESS: 'CASHBACK_SUCCESS',
    CASHBACK_FAILURE: 'CASHBACK_FAILURE',
    TOTAL_CASHBACKS_SUCCESS: 'TOTAL_CASHBACKS_SUCCESS'
}

export const getBalanceConstants = {
    GET_BALANCE_SUCCESS: 'GET_BALANCE_SUCCESS'
}