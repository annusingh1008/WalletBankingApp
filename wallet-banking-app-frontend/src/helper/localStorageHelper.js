const getLocal = (key) => {
  return localStorage.getItem(key);
};

export const getUserEmail = (email) => {
  return getLocal(email);
};
