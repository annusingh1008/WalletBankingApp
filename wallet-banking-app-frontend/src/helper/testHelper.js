import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const localStorageData = {
  email: "annu",
  password: "1234",
};

export const localStorageMock = (function () {
  return {
    getItem: function (key) {
      return localStorageData[key];
    },
    setItem: jest.fn(),
  };
})();

const mockStore = configureMockStore();

export const renderWithRedux = (ui, { store } = {}) => {
  return {
    ...render(<Provider store={mockStore(store)}>{ui}</Provider>),
    store,
  };
};
