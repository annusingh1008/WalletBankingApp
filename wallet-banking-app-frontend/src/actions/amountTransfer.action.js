import axios from "axios";
import { amountTransferConstants } from "./constants";

export const amountTransfer = (amountDetails) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/transfer/amount", {
      ...amountDetails,
    });

    if (res.status === 200) {
      if (res.data === "User does not exist") {
        alert("User does not exist! \nPlease check your email.");
      } else if (res.data === "Amount Transferred Successfully...!!") {
        alert("Transaction Successful");

        dispatch({
          type: amountTransferConstants.AMOUNT_TRANSFER_SUCCESS,
          payload: {
            amount:
              parseInt(amountDetails.currentAmount) -
              parseInt(amountDetails.amount),
            email: amountDetails.email,
          },
        });
      }
    }
  };
};
