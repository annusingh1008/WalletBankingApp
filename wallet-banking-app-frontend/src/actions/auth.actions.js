import axios from "axios";
import { authConstants } from "./constants";

export const login = (user) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/signin", {
      ...user,
    });

    if (res.status === 200) {
      if (res.data === "Signed in Successfully...!!") {
        localStorage.setItem("user", JSON.stringify(user));
        localStorage.setItem("email", user.email);

        dispatch({
          type: authConstants.LOGIN_SUCCESS,
          payload: user,
        });
      } else if (res.data === "Account does not exists with this email...!!") {
        alert("Account does not exists!! \nPlease check your email..");
      } else if (res.data === "Incorrect Password...!!") {
        alert("Please check your Password!!");
      }
    }
  };
};

export const getAuthDetails = (state) => state.auth;

export const signout = () => {
  return async (dispatch) => {
    dispatch({ type: authConstants.LOGOUT_REQUEST });
    localStorage.clear();
    dispatch({ type: authConstants.LOGOUT_SUCCESS });
  };
};
