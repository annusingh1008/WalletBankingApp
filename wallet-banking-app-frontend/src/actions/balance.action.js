import axios from "axios";
import { getBalanceConstants } from "./constants";

export const getUserBalance = (email) => {
  return async (dispatch) => {
    const res = await axios.get(`http://localhost:8085/getBalance/${email}`);

    dispatch({
      type: getBalanceConstants.GET_BALANCE_SUCCESS,
      payload: res.data,
    });
  };
};
