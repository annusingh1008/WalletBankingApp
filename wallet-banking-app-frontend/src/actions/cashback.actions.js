import axios from "axios";
import { cashbackConstants } from "./constants";

export const getAllCashBacks = (currentPage, email) => {
  return async (dispatch) => {
    const userEmail = localStorage.getItem("email");
    const res = await axios.get(
      `http://localhost:8085/getAllCashbacks/${userEmail}?pageNumber=${currentPage}&pageSize=${10}`
    );

    dispatch({
      type: cashbackConstants.CASHBACK_SUCCESS,
      payload: res.data,
    });
  };
};

export const getCashbacks = (state) => state.cashbacks;

export const getTotalCashbacks = (email) => {
  return async (dispatch) => {
    const res = await axios.get(
      `http://localhost:8085/getTotalCashbacks/${email}`
    );

    dispatch({
      type: cashbackConstants.TOTAL_CASHBACKS_SUCCESS,
      payload: res.data,
    });
  };
};
