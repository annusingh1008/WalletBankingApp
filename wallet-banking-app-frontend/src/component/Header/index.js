import React, { useState } from "react";
import { Navbar, Container, Nav, Modal, Button } from "react-bootstrap";
import { useDispatch } from "react-redux";
import { Link, NavLink } from "react-router-dom";
import { signout } from "../../actions/auth.actions";
import { getUserEmail } from "../../helper/localStorageHelper";

const Header = () => {
  const dispatch = useDispatch();
  const [confirmModal, setConfirmModal] = useState(false);

  const logout = () => {
    dispatch(signout());
  };

  const confirmSignout = (e) => {
    e.preventDefault();
    setConfirmModal(true);
  };

  const isUserLoggedIn = getUserEmail("email") !== null ? true : false;

  return (
    <Navbar
      collapseOnSelect
      fixed="top"
      expand="lg"
      bg="dark"
      variant="dark"
      style={{ zIndex: 2 }}
    >
      <Container fluid>
        <Link to="/" className="navbar-brand">
          Wallet Banking
        </Link>
      </Container>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        {!isUserLoggedIn ? (
          <Nav>
            <li className="nav-item">
              <NavLink to="signin" className="nav-link">
                Signin
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink to="signup" className="nav-link">
                Signup
              </NavLink>
            </li>
          </Nav>
        ) : (
          <Nav>
            <li className="nav-item">
              <span
                data-tesid="signout-btn"
                className="nav-link"
                onClick={confirmSignout}
              >
                Signout
              </span>
            </li>
          </Nav>
        )}

        {confirmModal && (
          <Modal
            data-testid="confirm-modal"
            show={confirmModal}
            // onHide={() => {
            //   setConfirmModal(false);
            // }}
            backdrop="static"
            keyboard={false}
          >
            <Modal.Header>
              <Modal.Title>Confirm</Modal.Title>
            </Modal.Header>
            <Modal.Body>Are you sure, you want to Signout</Modal.Body>
            <Modal.Footer>
              <Button
                data-testid="set-modal"
                variant="secondary"
                onClick={() => setConfirmModal(false)}
              >
                Close
              </Button>
              <Button
                data-testid="confirm-signout-btn"
                variant="primary"
                onClick={logout}
              >
                Confirm
              </Button>
            </Modal.Footer>
          </Modal>
        )}
      </Navbar.Collapse>
    </Navbar>
  );
};

export default Header;
