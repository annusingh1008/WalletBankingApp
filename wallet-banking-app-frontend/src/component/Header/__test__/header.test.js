import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { localStorageMock } from "../../../helper/testHelper";
import { signout } from "../../../actions/auth.actions";
import Header from "../index";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../store";
import { getUserEmail } from "../../../helper/localStorageHelper";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("react-bootstrap", () => {
  const Navbar = jest.fn(({ children }) => (
    <div data-testid="Navbar">{children}</div>
  ));
  const Container = jest.fn(({ children }) => (
    <div data-testid="Container">{children}</div>
  ));
  Navbar.Toggle = jest.fn(({ children }) => (
    <div data-testid="Toggle">{children}</div>
  ));
  Navbar.Collapse = jest.fn(({ children }) => (
    <div data-testid="Collapse">{children}</div>
  ));

  const Nav = jest.fn(({ children }) => <div data-tesid="Nav">{children}</div>);
  const Modal = jest.fn(({ children }) => (
    <div data-tesid="Modal">{children}</div>
  ));
  Modal.Header = jest.fn(({ children }) => (
    <div data-testid="Header">{children}</div>
  ));
  Modal.Title = jest.fn(({ children }) => (
    <div data-testid="Title">{children}</div>
  ));
  Modal.Body = jest.fn(({ children }) => (
    <div data-testid="Body">{children}</div>
  ));
  Modal.Footer = jest.fn(({ children }) => (
    <div data-testid="Footer">{children}</div>
  ));
  const Button = jest.fn((props) => (
    <button onClick={props.onClick} data-testid={`${props["data-testid"]}`}>
      {props.children}
    </button>
  ));

  return { Navbar, Container, Nav, Modal, Button };
});

jest.mock("react-router-dom", () => ({
  Link: jest.fn(({ children }) => <div data-testid="link">{children}</div>),
  NavLink: jest.fn(({ to, "data-testid": dataTestId, children, onClick }) => (
    <a href={to} onClick={onClick} data-testid={dataTestId}>
      {children}
    </a>
  )),
  BrowserRouter: jest.fn(({ children }) => (
    <div data-testid="BrowserRouter">{children}</div>
  )),
}));

jest.mock("../../../actions/auth.actions", () => ({
  signout: jest.fn(() => ({ type: "signout" })),
}));

jest.mock("../../../helper/localStorageHelper", () => ({
  getUserEmail: jest.fn(() => "email"),
}));

describe("Header", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders properly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("When clicks on signout button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId, queryByRole } = render(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>
      );

      const node = screen.getByText("Signout");
      fireEvent.click(node);
      expect(node).not.toBeNull();
      const confirmNode = queryByTestId("confirm-signout-btn");
      fireEvent.click(confirmNode);
      expect(mockDispatch).toBeCalledWith(signout());
      // const closeButton = queryByTestId("set-modal");
      // fireEvent.click(closeButton);
    });
  });

  describe("When clicks on close button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = render(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>
      );

      const node = screen.getByText("Signout");
      fireEvent.click(node);
      expect(node).not.toBeNull();

      const closeButton = queryByTestId("set-modal");
      fireEvent.click(closeButton);
    });
  });
});
