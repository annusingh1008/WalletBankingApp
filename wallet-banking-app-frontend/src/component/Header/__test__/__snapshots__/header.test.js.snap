// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
<div
  data-testid="BrowserRouter"
>
  <div
    data-testid="Navbar"
  >
    <div
      data-testid="Container"
    >
      <div
        data-testid="link"
      >
        Wallet Banking
      </div>
    </div>
    <div
      data-testid="Toggle"
    />
    <div
      data-testid="Collapse"
    >
      <div
        data-tesid="Nav"
      >
        <li
          class="nav-item"
        >
          <span
            class="nav-link"
            data-tesid="signout-btn"
          >
            Signout
          </span>
        </li>
      </div>
    </div>
  </div>
</div>
`;
